transformations/leet_letters/transformation.py

```diff
@@ -0,0 +1,55 @@
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Leet speak letter perturbation based on https://simple.wikipedia.org/wiki/Leet, excluding the space > 0.
+"""
+
+leet_letter_mappings = {
+    "!": "1",
+    "7": "1",
+    "A": "4",
+    "B": "8",
+    "C": "0",
+    "D": "0",
+    "E": "3",
+    "G": "6",
+    "I": "1",
+    "J": "9",
+    "L": "7",
+    "N": "11",
+    "O": "0",
+    "S": "5",
+    "T": "7",
+    "X": "8",
+    "Z": "2",
+    "b": "6",
+    "e": "3",
+    "g": "9",
+    "h": "4",
+    "j": "7",
+    "m": "3",
+    "o": "0",
+    "w": "3",
+    "y": "4",
+    "|": "1",
+    "Θ": "0",
+    "ε": "3",
+    "ω": "3",
+    "∈": "3",
+    "∩∩": "3",
+}
+
+
+class LeetLetters(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(self):
+        super().__init__()
+
+    def generate(self, sentence: str):
+        perturbed = ""
+        for letter in sentence:
```

Comment:
I really like this idea of using leet letters as substitutes. But I think it's better to not convert all the characters in a sentence to Leet - eg.a word like "63t333n" can be hard to figure out - maybe you could control some percentage of them using a configurable parameter.

Reply:
Good point - Max leet replacements in a sentence is now configurable and set to 50% by default.